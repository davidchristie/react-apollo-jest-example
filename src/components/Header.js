import React, { Component } from 'react'
import {
  Collapse,
  Nav,
  Navbar,
  NavbarToggler,
  NavbarBrand
} from 'reactstrap'

import NavItems from './NavItems'

export default class Header extends Component {
  constructor (props) {
    super(props)
    this.toggle = this.toggle.bind(this)
    this.state = {
      open: false
    }
  }

  render () {
    return (
      <Navbar color='faded' light toggleable>
        <NavbarToggler onClick={this.toggle} right />
        <NavbarBrand>React Apollo Jest Example</NavbarBrand>
        <Collapse isOpen={this.state.open} navbar>
          <Nav className='ml-auto' navbar>
            <NavItems />
          </Nav>
        </Collapse>
      </Navbar>
    )
  }

  toggle () {
    this.setState({
      open: !this.state.open
    })
  }
}
