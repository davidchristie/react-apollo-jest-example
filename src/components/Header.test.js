import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import React from 'react'
import { NavbarToggler } from 'reactstrap'

import Header from './Header'

describe('Header', () => {
  it('renders closed state correctly', () => {
    const output = shallow(<Header />)
    expect(toJson(output)).toMatchSnapshot()
  })

  it('renders open state correctly', () => {
    const output = shallow(<Header />)
    output.find(NavbarToggler).simulate('click')
    expect(toJson(output)).toMatchSnapshot()
  })
})
