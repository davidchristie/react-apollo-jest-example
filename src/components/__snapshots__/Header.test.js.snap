// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders closed state correctly 1`] = `
<Navbar
  color="faded"
  light={true}
  role="navigation"
  tag="nav"
  toggleable={true}
>
  <NavbarToggler
    onClick={[Function]}
    right={true}
    tag="button"
    type="button"
  />
  <NavbarBrand
    tag="a"
  >
    React Apollo Jest Example
  </NavbarBrand>
  <Collapse
    delay={
      Object {
        "hide": 350,
        "show": 350,
      }
    }
    isOpen={false}
    navbar={true}
    onClosed={[Function]}
    onOpened={[Function]}
    tag="div"
  >
    <Nav
      className="ml-auto"
      navbar={true}
      tag="ul"
    >
      <Apollo(NavItems) />
    </Nav>
  </Collapse>
</Navbar>
`;

exports[`Header renders open state correctly 1`] = `
<Navbar
  color="faded"
  light={true}
  role="navigation"
  tag="nav"
  toggleable={true}
>
  <NavbarToggler
    onClick={[Function]}
    right={true}
    tag="button"
    type="button"
  />
  <NavbarBrand
    tag="a"
  >
    React Apollo Jest Example
  </NavbarBrand>
  <Collapse
    delay={
      Object {
        "hide": 350,
        "show": 350,
      }
    }
    isOpen={true}
    navbar={true}
    onClosed={[Function]}
    onOpened={[Function]}
    tag="div"
  >
    <Nav
      className="ml-auto"
      navbar={true}
      tag="ul"
    >
      <Apollo(NavItems) />
    </Nav>
  </Collapse>
</Navbar>
`;
